URAP-charter/web_scraping
# updated docker compose file so that it works again!

The purpose of this branch is to allow the crawler to run successfully via Docker Compose.

Docker Compose is a service that allows quick spin-up and spin-down of containerized applications and services. In our case, we can use it to easily spin up a School Spider Crawler and a MongoDB for the data to live in. This is especially good for demonstration purposes, as it is easy to run and easy to demonstrate.

There were some networking issues with connecting the crawler's container with the mongodb container, which was resulting in nothing being written to the db. This has been resolved by adding the "expose" line to the mongodb container (which exposes the mongo default 27017 port) and the "link" line to the crawler container (which allows you to use the name of another container as its pseudo-IP address).

To test this branch:
1. Ensure that in settings.py, the MONGO_URI parameter is set to "mongodb://mongodb_container:27017"
2. For easier testing, split the charter_schools... tsv file into 1000-line blocks: 
From the directory with the file, use the command: "split -l 1000 charter_school..."
3. In run_schoolspider.py, change the run scrapy command to use the file "xaa" in place of "charter_school..." (the rest of the path should be the same)
4. Navigate to the root directory, where the docker-compose.yml file lives and use the command "docker-compose up --build"
5. Once the crawler finishes, use another terminal to access the Mongo container: "docker exec -it mongodb_container bash"
6. From there, use the command "mongo" to enter the mongo CLI
7. In the Mongo CLI, use the commands "show dbs", "use schoolSpider", and "db.outputItems.count()" to verify that a couple of items have been saved.
